pyproject.toml

```diff
@@ -1,3 +1,43 @@
+[tool.ruff]
+exclude = [
+   ".git",
+   "bootloader",
+   "build",
+   "dist",
+   "doc/_build",
+   "doc/conf.py",
+   "PyInstaller/lib",
+   "tests/functional/data/name_clash_with_entry_point",
+   "tests/functional/data/sphinx/conf.py",
+   "tests/unit/test_modulegraph",
+]
+ignore = ["PLC1901"]
+line-length=120
+select = [
+  "C90",     # McCabe cyclomatic complexity
+  "DTZ",     # flake8-datetimez
+  "E",       # pycodestyle
+  "F",       # Pyflakes
+  "INT",     # flake8-gettext
+  "PLC",     # Pylint conventions
+  "PLE",     # Pylint errors
+  "PLR091",  # Pylint refactor just for max-args, max-branches, etc.
+  "PYI",     # flake8-pyi
+  "T10",     # flake8-debugger
+  "W",       # pycodestyle
+  "YTT",     # flake8-2020
```

Comment:
A lot of these look like they'll end up irritating me pretty quickly. Let's get rid of McCabe since we already know that PyInstaller's code is overcomplicated and having to configure its thresholds so that it accepts that mess removes the purpose of the check. My experience of `pylint` has been zero signal and a lot of noise so let's be rid of those ones too. `flake8-debugger`, `flake8-2020` and `flake8-gettext` all sound like poor substitutes for the coarsest of code reviews. `flake8-pyi` is not much use to us given that we don't ship pyi files (and have no intention of doing so).